commons/docker/api.go

```diff
@@ -100,7 +100,9 @@ func NewContainer(cd *ContainerDefinition, start bool, timeout time.Duration, on
 	if useRegistry {
 		if err := PullImage(iid.String()); err != nil {
 			glog.V(2).Infof("Unable to pull image %s: %v", iid.String(), err)
-			return nil, err
+			if _, err2 := lookupImage(iid.String()); err2 != nil {
+				return nil, err2
+			}
 		}
```

Comment:
Don't implode just because we can't pull; only implode if we can't pull AND we don't have the image locally.